server/src/optima/project.py

```diff
@@ -615,7 +615,15 @@ def uploadExcel():
         D.G.workbookname = D.G.projectname + '.xlsx'
         D.G.inputprograms = deepcopy(project.programs)
         D.G.inputpopulations = deepcopy(project.populations)
-        D = updatedata(D, input_programs = project.programs, savetofile = False)
+
+        # Is this the first time? if so then we have to run simulations
+        try:
+            D.S # the cool kids don't reRun when they already have simulations
+            shouldReRun = False
+        except AttributeError:
+            shouldReRun = True
```

Comment:
I think "Easier to ask for forgiveness than permission" shouldn't be applied to `S`. Why?

Let's take `D` for example. We expect it to be defined always. There shouldn't be a project without a model (=D). In this case we don't want to always check for D. If project exists D exists & you can operate like that. Still you never know for sure & that's why it's important to wrap the whole accessing in a try-catch (which Flask basically does with every request).

Let's have a closer look at `S`. S is not expected to be there always. In fact we know for a certain common case S won't be defined. The information is important to enable a different code path. I would say it's even not 'LBYL', because we don't make any call on S. It's just checking a state & acting on it.

Does that make sense?